tracer/src/Datadog.Trace/Iast/Propagation/DefaultInterpolatedStringHandlerModuleImpl.cs

```diff
@@ -51,7 +55,16 @@ public static unsafe void FullTaintIfAnyTainted(IntPtr target, string? input)
             var rangesResult = new[] { new Range(0, 0, tainted!.Ranges[0].Source, tainted.Ranges[0].SecureMarks) };
             if (!targetIsTainted)
             {
-                taintedObjects.Taint(target, rangesResult);
+                // Safe guard to avoid memory leak
+                if (_taintedRefStructs.Count > 16)
+                {
+                    _taintedRefStructs.Clear();
```

Comment:
_Clearing_ here seems, curious 🤔 Doesn't that risk that you lose a bunch of stuff. There's no _plausible_ way you should leak here anyway I think, right? Essentially it would be a programming error?